### ISSUE TITLE
Triggering on-demand backup returns error

### ISSUE THREAD

**Issue by mdhowle** [ansible/ansible_collections_azure#1522]
##### SUMMARY
Attempting to trigger an on-demand backup of VM returns
`"Create or Updating fail, no match message return, return info as None"`

It does appear the snapshots are occurring.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`azure.azcollection.azure_rm_backupazurevm`

##### ANSIBLE VERSION

```paste below
ansible [core 2.16.5]
  config file = /home/XXXX/.ansible.cfg
  configured module search path = ['/home/XXXX/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/XXXX/ansible/venv/ansible-azure/lib/python3.11/site-packages/ansible
  ansible collection location = /home/XXXX/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/XXXX/ansible/venv/ansible-azure/bin/ansible
  python version = 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0] (/home/XXXX/ansible/venv/ansible-azure/bin/python)
  jinja version = 3.1.3
  libyaml = True

```

##### COLLECTION VERSION

```paste below
ansible-galaxy collection list azure.azcollection

# /home/XXXX/.ansible/collections/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 2.3.0

# /home/XXXX/ansible/venv/ansible-azure/lib/python3.11/site-packages/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 1.19.0
```

##### CONFIGURATION

```paste below
CACHE_PLUGIN(/home/XXXX/.ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/home/XXXX/.ansible.cfg) = ~/.ansible/facts/
CACHE_PLUGIN_TIMEOUT(/home/XXXX/.ansible.cfg) = 0
CONFIG_FILE() = /home/XXXX/.ansible.cfg
DEFAULT_GATHERING(/home/XXXX/.ansible.cfg) = smart
DEFAULT_TIMEOUT(/home/XXXX/.ansible.cfg) = 60
EDITOR(env: EDITOR) = vim
GALAXY_SERVER_LIST(/home/XXXX/.ansible.cfg) = ['rh_automation_hub']
HOST_KEY_CHECKING(/homeXXXX/.ansible.cfg) = False

```

##### OS / ENVIRONMENT

Debian 12 
Python 3.11.2

##### STEPS TO REPRODUCE

```yaml

---
- name: Create a backup
  hosts: all
  tasks:
    - name: Create a backup of an Azure VM
      azure.azcollection.azure_rm_backupazurevm:
        resource_group: XXXX
        recovery_vault_name: XXXX
        resource_id: '/subscriptions/XXXX/resourceGroups/XXXX/providers/Microsoft.Compute/virtualMachines/XXXX'
        backup_policy_id: '/subscriptions/XXXX/resourceGroups/XXXX/providers/Microsoft.RecoveryServices/vaults/XXXX/backupPolicies/DefaultPolicy'
        state: 'backup'
```

##### EXPECTED RESULTS

VM backup triggered, or at least a better error message.

##### ACTUAL RESULTS

[
```
TASK [Create a backup of an Azure VM] ********************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Create or Updating fail, no match message return, return info as None"}
```
Verbose: https://gist.github.com/mdhowle/0b23073bcd8a2aa7b12d3f55db43d313

I am using SP credentials but I have also tried my personal credentials with the same result. Running  `az backup protection backup-now` works with SP credentials.

**Comment by Fred-sun** [comment]
@mdhowle Yes, there is no return value after triggering the backup now, not need to format the return value, it has been fixed in PR #1531 . Thank you!

**Comment by mdhowle** [comment]
Shouldn't triggering an on-demand backup return a reference to the backup job? Then I can, ideally, use it to obtain the job details (see #1127)
